proto/proxy/destination/destination.proto

```diff
@@ -78,8 +78,24 @@ message WeightedAddr {
   common.TcpAddress addr = 1;
   uint32 weight = 3;
   map<string, string> metric_labels = 4;
+  TlsVerification tls_verification = 5;
+}
+
+// Which strategy should be used for verifying TLS.
+message TlsVerification {
+  oneof strategy {
+    K8sPodNamespace k8s_pod_namespace = 1;
+  }
+
+  // Verify the certificate based on the Kubernetes pod name, and ensure
+  // that the pod is configured with the same Conduit control plane
+  // namespace as this proxy.
+  message K8sPodNamespace {
+    string namespace = 1;
+    string pod_name = 2;
```

Comment:
We need the Conduit controller namespace so we know whether the pod is in the same mesh. I suggest we have two fields, `pod_namespace` and `controller_namespace` to make that clear.

Reply:
Just so we're clear, do we need both the pod's namespace and_the pod's controller namespace? I thought we needed the pod's _name_ and the pod's controller namespace?

Reply:
pod names aren't guaranteed to be unique across namespaces. In any case, I think we should always be explicit about what kind of namespace we're referring to. It's a very overloaded term, which quickly gets confusing...

Reply:
Yeah, I was planning on renaming the field, I just wanted to make sure the message has all the fields necessary as well.

Reply:
Changed in a7427fb51b287f888b07f23ddd1d24d2aca2b173.